integrations/indexers/covalent.md

```diff
@@ -0,0 +1,160 @@
+
+
+The Covalent API will return a list of token holders for the ERTH token. Unless you already owned some ERTH tokens, your address will be missing from that list. Head over to the [Moonbase Alpha ERC-20 Faucet](https://moonbase-minterc20.netlify.app/) to generate some ERTH tokens for yourself. Now repeat the same Covalent API request as above. You should now see your address in the list of token holders for the ERTH token.
+
+## Javascript Examples
+Copy and paste the below code block into your preferred environment, or [JSFiddle](https://jsfiddle.net/). After setting the API key, set the address constant. Remember our chain ID is ‘1287’ for Moonbase Alpha.
+
+=== "Using Fetch"
+    ```js
+    // set your API key
+	const APIKEY = 'YOUR API KEY HERE';
+
+	function getData() {
+    const address = '0xFEC4f9D5B322Aa834056E85946A32c35A3f5aDD8'; //example
+    const chainId = '1287'; //Moonbeam Testnet (Moonbase Alpha Chain ID)
+    const url = new URL(`https://api.covalenthq.com/v1/${chainId}/address/${address}/balances_v2/`);
+    
+    url.search = new URLSearchParams({
+        key: APIKEY
+    })
+
+    // use fetch API to get Covalent data
+    fetch(url)
+    .then((resp) => resp.json())
+    .then(function(data) {
+        const result = data.data;
+  
+        console.log(result)
+        return result
+        }
+	)}
+
+    getData();
+    ```
+
+=== "Using Async"
+    ```js
+    // set your API key
+    const APIKEY = 'YOUR API KEY HERE';
+	const address = '0xFEC4f9D5B322Aa834056E85946A32c35A3f5aDD8'; //example
+	const chainId = '1287'; //Moonbeam Testnet (Moonbase Alpha Chain ID)
+	const url = new URL(`https://api.covalenthq.com/v1/${chainId}/address/${address}/balances_v2/`);
+
+    url.search = new URLSearchParams({
+        key: APIKEY
+    })
+
+    async function getData() {
+    	const response = await fetch(url);
+    	const result = await response.json();
+    	console.log(result)
+    	return result;
+	}
+
+	getData();
+
+    ```
+
+The output should resemble the below. The balances endpoint returns a list of all ERC20 and NFT token balances including ERC721 and ERC1155 along with their current spot prices (if available).
+
+![Javascript Console Output](/images/covalent/covalentjs.png)
+
+## Python Example
+Covalent doesn’t have an official API wrapper. To query the API directly you will have to use the Python [requests library](https://pypi.org/project/requests/). Install requests into your environment from the command line with `pip install requests`. Then import it and use it in your code. Use the HTTP verbs get methods to return the information from the API. Copy and paste the below code block into your preferred environment and run it. 
+
+```python
+import requests
+
+def fetch_wallet_balance(address):
+	api_url = 'https://api.covalenthq.com'
+    endpoint = f'/v1/1287/address/{address}/balances_v2/'
+    url = api_url + endpoint
+    r = requests.get(url, auth=('YOUR API KEY HERE',''))
```

Comment:
what is the second parameter for? in `auth=('YOUR API KEY HERE','')`

Reply:
It's actually empty! But I can clarify that in the text.